www/js/app.js

```diff
@@ -64,34 +65,100 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     
+    /**
+     * Apply any override parameters that might be in the querystring.
+     * This is used for communication between the PWA and any local code (e.g. Firefox Extension), both ways.
+     * It is also possible for DEV (or user) to launch the app with certain settings, or to unset potentially
+     * problematic settings, by crafting the querystring appropriately.
+     */
+    (function overrideParams() {
+        var rgx = /[?&]([^=]+)=([^&]+)/g;
+        var matches = rgx.exec(window.location.search);
+        while (matches) {
+            if (matches[1] && matches[2]) {
+                var paramKey = decodeURIComponent(matches[1]);
+                var paramVal = decodeURIComponent(matches[2]);
+                if (paramKey !== 'title') {
+                    console.debug('Setting key-pair: ' + paramKey + ':' + paramVal);
+                    // Make values Boolean if 'true'/'false'
+                    paramVal = paramVal === 'true' || (paramVal === 'false' ? false : paramVal);
+                    settingsStore.setItem(paramKey, paramVal, Infinity);
+                    params[paramKey] = paramVal;
+                }
+            }
+            matches = rgx.exec(window.location.search);
+        }
+        // If we are in an extension, send a 'success' message to the extension
+        if (params.extensionURL && ~window.location.href.indexOf(params.PWAServer)) {
+            var message = '?PWA_launch=success';
+            // DEV: To test failure of the PWA, you could pause on next line and set message to '?PWA_launch=fail'
+            // Note that, as a failsafe, the PWA_launch key is set to 'fail' (in the extension) before each PWA launch
+            // so we need to send a 'success' message each time the PWA is launched
+            var frame = document.createElement('iframe');
+            frame.id = 'kiwixComm';
+            frame.style.display = 'none';
+            document.body.appendChild(frame);
+            frame.src = params.extensionURL + '/www/index.html'+ message;
+            // Now remove redundant frame. We cannot use onload, because it doesn't give time for the script to run.
+            setTimeout(function () {
+                var kiwixComm = document.getElementById('kiwixComm');
+                // The only browser which does not support .remove() is IE11, but it will never run this code
+                if (kiwixComm) kiwixComm.remove();
+            }, 3000);
+        }
+    })();
+
+    /**
+     * Set the State and UI settings associated with parameters defined above
+     */
+    document.getElementById('hideActiveContentWarningCheck').checked = params.hideActiveContentWarning;
+    document.getElementById('showUIAnimationsCheck').checked = params.showUIAnimations;
+    document.getElementById('titleSearchRange').value = params.maxSearchResultsSize;
+    document.getElementById('titleSearchRangeVal').innerHTML = params.maxSearchResultsSize;
```

Comment:
To avoid the code scanning warning, you could escape the content before putting it in `.innerHTML` (even if, in our case, we know it will always be a number)